dm: update precheck (#10977)
pingcap/docs: dm/dm-precheck.md

- Compatibility of the upstream MySQL table schema

    - Check whether the upstream tables have foreign keys, which are not supported by TiDB. A warning is returned if a foreign key is found in the precheck.

>     - Check whether there are compatibility differences in character sets. For more information, see [TiDB Supported Character Sets](/character-set-and-collation.md).

Reviewer: ```suggestion
    - Check whether the upstream tables have compatibility differences in character sets. For more information, see [TiDB Supported Character Sets](/character-set-and-collation.md).
```

    - Check whether the upstream tables have primary key constraints or unique key constraints (introduced from v1.0.7).

    > **Warning:**
    >

>     > - When the upstream uses incompatible character sets, you can still continue the replication by creating tables with the utf8mb4 character set in the downstream. However, this practice is not recommended. You are advised to change the character set used by the upstream or use the same character set in both the upstream and downstream.

Author: ```suggestion
    > - When the upstream uses incompatible character sets, you can still continue the replication by creating tables with the utf8mb4 character set in the downstream. However, this practice is not recommended. You are advised to replace the character set used by the upstream with another character set that is supported in downstream.
```

>     > - When the upstream uses incompatible character sets, you can still continue the replication by creating tables with the utf8mb4 character set in the downstream. However, this practice is not recommended. You are advised to replace the character set used by the upstream with another character set that is supported in downstream.

Reviewer: ```suggestion
    > - When the upstream uses incompatible character sets, you can still continue the replication by creating tables with the utf8mb4 character set in the downstream. However, this practice is not recommended. You are advised to replace the incompatible character set used by the upstream with another character set that is supported in downstream.
```

>     > - When the upstream tables have no primary key constraints or unique key constraints, the same row of data might be replicated multiple times to the downstream, which might affect the performance of replication. In a production environment, it is recommended that you specify primary key constraints or unique key constraints.

Reviewer: ```suggestion
    > - When the upstream tables have no primary key constraints or unique key constraints, the same row of data might be replicated multiple times to the downstream, which might affect the performance of replication. In a production environment, it is recommended that you specify primary key constraints or unique key constraints for the upstream database.
```

>     > - When the upstream tables have no primary key constraints or unique key constraints, the same row of data might be replicated multiple times to the downstream, which might affect the performance of replication. In a production environment, it is recommended that you specify primary key constraints or unique key constraints for the upstream database.

Reviewer 2: 性能下降和同步多次是并列关系的两个事情；主键是在表上而不是数据库上。
```suggestion
    > - When the upstream tables have no primary key constraints or unique key constraints, the same row of data might be replicated multiple times to the downstream, which might also affect the performance of replication. In a production environment, it is recommended that you specify primary key constraints or unique key constraints for the upstream table.
```

    - REPLICATION CLIENT permission
    - REPLICATION SLAVE permission

* Database primary-secondary configuration

>     - It is recommended that you specify the database ID `server_id` for the upstream database (GTID is recommended for non-AWS Aurora environments), so as to avoid primary-secondary replication failures.

Reviewer: ```suggestion
    - To avoid primary-secondary replication failures, it is recommended that you specify the database ID `server_id` for the upstream database (GTID is recommended for non-AWS Aurora environments).
```
